src/field_theory/adjoin.lean

```diff
@@ -250,10 +252,39 @@ adjoin_adjoin_left _ _ _
 lemma adjoin_simple_comm (β : E) : ↑F⟮α⟯⟮β⟯ = (↑F⟮β⟯⟮α⟯ : intermediate_field F E) :=
 adjoin_adjoin_comm _ _ _
 
+lemma adjoin_simple_to_subalgebra_of_integral (hα : is_integral F α) :
+  (F⟮α⟯).to_subalgebra = algebra.adjoin F {α} :=
```

Comment:
It seemed to me that this proof could be a lot simpler if we combined `subalgebra.is_field_of_algebraic`, but apparently it isn't so easy to show `is_field (S : subalgebra F E) ->∀ x, x ∈ S -> x⁻¹ ∈ S`. So there are two options here: you can improve the usefulness of `is_field` and use `subalgebra.is_field_of_algebraic`, or keep the proof as is (and someone can make `is_field` easier to work with in a later PR).

Reply:
To me, it even seems hard to use `subfield.is_field_of_algebraic` because `algebra.adjoin F {α} : subalgebra F E` but E/F is not necessarily algebraic.

Reply:
Really you want to know that a subalgebra of an algebraic subalgebra is algebraic.